### PR TITLE
Support for DSF Files.

### PR DESCRIPTION
**Pull request by docbobo**
This PR address support for DSF files as discussed in #155. Implementation is based on information available from Sony's DSF File Format Specification.

Since this is my first stab as Python, there might be a few rough edges. Please feel free to point them out and I will happily improve them.

### CODE REVIEW

File: mutagen/dsf.py [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+
+
+__all__ = ["DSF", "Open", "delete"]
+
+
+class error(MutagenError):
+    pass
+
+
+class DSFHeaderError(error):
```
Comment (lazka) [inline mutagen/dsf.py]: Is there a need for a separate exception type? I'd prefer if just error is used.

File: mutagen/dsf.py [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+
+        self.fileobj.seek(self.chunk_offset)
+        self.fileobj.write(f.getvalue())
+
+    def pprint(self):
+        return u"DSD Chunk (Total file size = %d, Pointer to Metadata chunk = %d)" % (
+            self.total_size, self.offset_metdata_chunk)
+
+
+class FormatChunk(DSFChunk, StreamInfo):
```
Comment (lazka) [inline mutagen/dsf.py]: Please create a separate StreamInfo class (maybe taking the FormatChunk in __init__). This way the DSFChunk/FormatChunk API isn't exposed to the user.

File: mutagen/dsf.py [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+            fileobj.seek(dsd_header.offset_metdata_chunk)
+            fileobj.write(data)
+            fileobj.truncate()
+
+            # Update total file size
+            dsd_header.total_size = fileobj.tell()
+            dsd_header.write()
+
+        finally:
+            fileobj.close()
```
Comment (lazka) [inline mutagen/dsf.py]: Don't close fileobj. That's handled by loadfile() if a filename is passed.

File: mutagen/dsf.py (hunk 1 above)
Comment (docbobo) [inline mutagen/dsf.py]: This was inspired by what I found in flac.py. I believe you are asking whether there is value in differentiating between general exceptions and header exceptions - which there probably isn't. Will rework.

File: mutagen/dsf.py (hunk 2 above)
Comment (docbobo) [inline mutagen/dsf.py]: Will have look.

File: mutagen/dsf.py (hunk 3 above)
Comment (docbobo) [inline mutagen/dsf.py]: Sure thing.

**Comment by docbobo** [comment]
Apologies for the rebase, but I had to change the author info - was configured improperly.

**Comment by docbobo** [comment]
I believe I've incorporated all your suggestions. Let me know if there's anything else.

**Comment by lazka** [comment]
Thanks.

**Comment by audiomuze** [comment]
Does this mean Mutagen now supports tagging of DSF files?

**Comment by lazka** [comment]
@audiomuze yes, since 1.37.

**Pull request merged**